source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -186,7 +174,7 @@ public RemoteIdentity ReadRemoteIdentity()
             {
                 log.Write(EventType.Error, "Response:");
                 log.Write(EventType.Error, line);
-                log.Write(EventType.Error, streamReader.ReadToEnd());
+                log.Write(EventType.Error, new StreamReader(stream, new UTF8Encoding(false)).ReadToEnd());
```

Comment:
I wonder if a short timeout should be set here, in this error case we don't know how much has been sent over the wire. Potentially we could block for a long time here. In a sense this is now more likely since we consume the `emptyLine` that is sent after the identity line, so in the case the identity line is wrong we could hang.

Reply:
My understanding is that `ReadToEnd` on a network stream will block forever until the connection's closed or the read timeout is reached.

Reply:
I think that is true, so I guess what it does now is ok.

Reply:
It's not any different than before. I think we can put this in and if need be deal with any blocking issues that arise.